sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -231,7 +218,18 @@ private[sbtplugin] object ScalaJSPluginInternal {
             Set(output, sourceMapFile)
           } (realFiles.toSet)
 
-          Attributed.blank(output).put(scalaJSSourceMap, sourceMapFile)
+          /* This is is somewhat not nice: We must make assumptions about the
+           * specification of the specific linker in use. Otherwise, we do not
+           * know how to interpret / load the resulting file.
+           *
+           * See jsEnvInput (via scalaJSLinkedFile) to see how this is used.
+           */
+          val moduleKind =
+            (scalaJSLinkerConfig in key).value.moduleKind
```

Comment:
This should be moved above the `Def.task`. See the big comment that starts with "It is very important that" :p